app/Http/Controllers/Assets/BulkAssetsController.php

```diff
@@ -208,61 +208,86 @@ public function update(Request $request)
+                                    if ($decrypted_old != $this->update_array[$field->db_column]) {
+                                        $asset->{$field->db_column} = \Crypt::encrypt($this->update_array[$field->db_column]);
+                                    } else {
+                                        \Log::debug('The decrypted existing value is the same as the one submitted - unset it');
+                                        unset($this->update_array[$field->db_column]);
+                                        unset($asset->{$field->db_column});
+                                    }
+
+                                // These fields aren't encrypted, just carry on as usual
+                                } else {
+
+
+                                    if ((array_key_exists($field->db_column, $this->update_array)) && ($asset->{$field->db_column} != $this->update_array[$field->db_column])) {
+
+                                        // Check if this is an array, and if so, flatten it
+                                        if (is_array($this->update_array[$field->db_column])) {
+                                            $asset->{$field->db_column} = implode(', ', $this->update_array[$field->db_column]);
+                                        } else {
+                                            $asset->{$field->db_column} = $this->update_array[$field->db_column];
                                         }
-                                        continue; 
-                                } else { 
-                                        $array = $this->update_array; 
-                                        array_except($array, $field->db_column); 
-                                        $asset->save($array); 
-                                    }     
-                                    if (!$asset->save()) {
-                                        $error_bag[] = $asset->getErrors();
-                                } 
-                    } 
-                }
-                } else {
-                    Asset::find($assetId)->update($this->update_array);
-                } 
-            } 
-            if(!empty($error_bag)) {
-                $errors = [];
-               //find the customfield name from the name of the messagebag items 
-                foreach ($error_bag as $key => $bag) {
-                  foreach($bag->keys() as $key => $value) {
-                        CustomField::where('db_column', $value)->get()->map(function($item) use (&$errors) {
-                            $errors[] = $item->name; 
-                        });
+                                    }
+                                }
+
+                            } // endforeach
+                    } // end custom field check
+                } // end custom fields handler
+
+
+
+                // Check if it passes validation, and then try to save
+                if (!$asset->update($this->update_array)) {
```

Comment:
I think this is Laravel being confusing - but in my head the `->update()` method means "change the values on the actual object, but don't fire `save()` yet. But I might be mistaken because in the context of bulk actions, it's just a straight SQL `UPDATE` query. So if this *does* work as intended, my bad. But my concern here is that the Watson Validating trait won't have the opportunity to fire yet because it's not doing the `save()`? So it will never fail this `if()` clause. But, again, I could be mistaken.

Reply:
It definitely catches on validation